specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,963 @@
+
+The adaptive profile extends the base profile with additional instructions such
+as `br` (**Bullet 4**) and the additional opt-in instructions or intrinsics in
+(**Bullet 5**)
+
+See also the section on [data types and values](#data-types-and-values) for more
+information about the creation and usage of LLVM values.
+
+## Run-Time Functions
+
+The following run-time functions must be supported by all backends:
+
+| Function                            | Signature            | Description                                                                                                                                                                                                                                                  |
+| :---------------------------------- | :------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| __quantum__rt__initialize           | `void(i8*)`          | Initializes the execution environment. Sets all qubits to a zero-state if they are not dynamically managed.                                                                                                                                                 |
+| __quantum__rt__tuple_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of a tuple and how many tuple elements it has. The second parameter defines a string label for the tuple. Depending on the output schema, the label is included in the output or omitted.  |
+| __quantum__rt__array_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of an array and how many array elements it has. The second parameter defines a string label for the array. Depending on the output schema, the label is included in the output or omitted. |
+| __quantum__rt__result_record_output | `void(%Result*,i8*)` | Adds a measurement result to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                         |
+| __quantum__rt__bool_record_output   | `void(i1,i8*)`       | Adds a boolean value to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                              |
+
+The following output recording functions can appear if you opt into supporting
+real-time integer calculations on an integer or fixed-point `iN` type (the profile compliant program sets the following module flag `!{i32 1, !"classical_ints", i32 N}` or  `!{i32 1, !"classical_fixed_points", i32 N}`).
```

Comment:
I think fixed point needs its own ouput recording function as we need a radix in order to interpret the integer value and we can't rely on tagging for interpetation.